CalibTracker/SiPixelESProducers/plugins/SiPixelFakeGenErrorDBObjectESSource.cc

```diff
@@ -35,9 +34,9 @@ std::unique_ptr<SiPixelGenErrorDBObject> SiPixelFakeGenErrorDBObjectESSource::pr
   //  open the GenError file(s)
   for (m = 0; m < obj->numOfTempl(); ++m) {
     edm::FileInPath file(GenErrorCalibrations_[m].c_str());
-    tempfile = (file.fullPath()).c_str();
+    std::string tempfile = file.fullPath();
```

Comment:
@iarspider , isn't it better/faster to create `std::string tempfile` outside the for loop?

Reply:
@Dr15Jones @makortel , is it intentional that `FileInPath::fullPath` return a copy and not a `const &`?

Reply:
Not sure. If we really care about repeated (de)allocation of memory, maybe change `fullPath` to return reference?

Reply:
> is it intentional that `FileInPath::fullPath` returns a copy and not a `const string&`?

I'd guess "not intentional". Or, I can't think of any good reason why it returns a copy and not a const reference.

Reply:
@iarspider It would be great if you could do it. On the same go I'd suggest to change `relativePath()` to return `const string&` as well, and remove the explicit definitions of copy constructor and assignment operator, and destructor.